specs/agents/tracing-sampling.md

```diff
@@ -7,16 +7,39 @@ Head-based sampling is where a sampling decision is made at the root of the dist
 before the details or outcome of the trace are known,
 and propagated throughout the trace.
 
-##### Configuration
+##### `transaction_sample_rate` configuration
 
-By default all transactions will be sampled.
+By default, all transactions will be sampled.
 Agents can be configured to sample probabilistically,
-by specifying a sampling probability in the range \[0,1\] using the configuration `ELASTIC_APM_TRANSACTION_SAMPLE_RATE`.
+by specifying a sampling probability in the range \[0,1\].
 e.g.
 
- - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=1` means all transactions will be sampled (the default)
- - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0` means no transactions will be sampled
- - `ELASTIC_APM_TRANSACTION_SAMPLE_RATE=0.5` means approximately 50% of transactions will be sampled
+ - `1` means all transactions will be sampled (the default)
+ - `0` means no transactions will be sampled
+ - `0.5` means approximately 50% of transactions will be sampled
+
+The maximum precision of the sampling rate is `0.001` (0.1%).
```

Comment:
This may be perceived as a breaking change. How likely is it that someone only wants to sample 1 in 10,000 requests?

Reply:
Sampling less than 0.1% seems unlikely to me, but that's based only on my experience and intuition. Maybe @roncohen or @nehaduggal can provide some input based on real-world usage. I think it would be OK to go to 0.01% if we're concerned it's borderline.

Reply:
+1 to got the extra flexibility from my perspective. I didn't see a downside to enable it.

Reply:
Are you saying you’d prefer the max precision to be 0.001%?

The only downside is that it would increase the max size of the tracestate Header by a little: `es=s:0.0001` vs `es=s:0.001`.
But yeah, doesn’t cost much and should get us on the safe side.

Reply:
OK - I agree, let's go with 0.0001.